packages/itwin/property-grid/src/components/PropertyGrid.tsx

```diff
@@ -171,8 +172,37 @@ export const PropertyGrid = ({
           await customOnDataChanged(dataProvider);
         }
       }
+      if(persistNullValueToggle) {
+        let res = await getShowNullValues();
+        if(res !== undefined && dataProvider)
+        {
+          dataProvider.includeFieldsWithNoValues = res;
+          setShowNullValues(res);
+        }        
+    }
     };
 
+    const getShowNullValues = async () => {
+      const userPrefs = IModelApp.userPreferences;
+      if(userPrefs) {
+        const accessToken = await IModelApp.getAccessToken();
+        const iModel = UiFramework.getIModelConnection();
+        const iTwinId = iModel?.iTwinId;
+        const iModelId = iModel?.iModelId;
+        const showNullValuesRes = await userPrefs.get({
```

Comment:
it would be much cleaner to move all the code related to getting and setting the null value toggle in the user preferences to another file so you can simply invoke it here, it always needs to have try / catch handling and logging in place for the error cases

Reply:
I can add the try/catch and use the logger from core. Based on the rest of the file, it seems weird to only move out the two functions I wrote to their own file. This same file already handles adding/removing favorites, building it's own context menu, etc. If we're gonna refactor it then that's fine, but it just seems weird to move this one thing out. I can if I need to though.